control/statesp.py

```diff
@@ -93,9 +93,15 @@ def _ssmatrix(data, axis=1):
     # Convert the data into an array or matrix, as configured
     # If data is passed as a string, use (deprecated?) matrix constructor
     if config.defaults['statesp.use_numpy_matrix'] or isinstance(data, str):
-        arr = np.matrix(data, dtype=float)
+        if np.isrealobj(data):
+            arr = np.matrix(data, dtype=float)
+        elif np.iscomplexobj(data):
+            arr = np.matrix(data, dtype=complex)
     else:
-        arr = np.array(data, dtype=float)
+        if np.isrealobj(data):
+            arr = np.array(data, dtype=float)
+        elif np.iscomplexobj(data):
+            arr = np.array(data, dtype=complex)
```

Comment:
Is this really necessary? Why not just use `matrix()` and `asarray()` without `dtype` and figure out the `dtype` later?

Reply:
Turns out if dtype is omitted completely, it could stell be `int` and trip operations later on.

The test suite already has complex casting warnings from Scipy and Slycot functions returning complex arrays (with 0 imaginary part), which are passed to `_ssmatrix`. Thus, my approach in  #438 is to use `dtype=np.result_type(np.float, arr)`, which ensures to cast from int to at least float and retains complex if arr is already complex.